### PR TITLE
Added task 6

### PR DESCRIPTION
**Pull request by ghiffari90**

### CODE REVIEW

**Review by nc-john-hutcheon — COMMENTED** [review]
Hey Ghiffari, some great code and tests here, but we're just missing one more test:

- What would happen if we went to an endpoint that was a valid article, but had no comments? (Have a think about yesterday's lecture content as to how you can handle this!)

**Comment by ghiffari90** [comment]
Hey John, I've realised it and it took me a while to make the test. That said, I've added the test for it in another pull request. Thanks for your feedback!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -116,3 +116,44 @@ describe("GET /api/articles/:article_id", () => {
...
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments }  }) => {
+        expect(Array.isArray(comments)).toBe(true);
+        expect(comments.length).not.toBe(0);
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            body: expect.any(String),
+            article_id: expect.any(Number),
```
Comment (nc-nathan-blanch) [inline __tests__/app.test.js:132]: any `article id`?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -116,3 +116,44 @@ describe("GET /api/articles/:article_id", () => {
...
+        });
+        expect(comments).toBeSortedBy('created_at', { descending: true });
+      })
+  });
+
+  test("400: Responds with a message 'Invalid id type' if the article_id in the URL is invalid", () => {
+    return request(app)
+      .get('/api/articles/one/comments')
+      .expect(400)
+      .then(({ body: { msg } }) => {
```
Comment (nc-nathan-blanch) [inline __tests__/app.test.js:146]: good test 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -116,3 +116,44 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(msg).toBe('Invalid id type')
+      });
+  });
+
+  test("404: Responds with a message 'not found' if the article_id in the URL is non-existent", () => {
+    return request(app)
+      .get('/api/articles/100/comments')
+      .expect(404)
+      .then(({ body: { msg } }) => {
```
Comment (nc-nathan-blanch) [inline __tests__/app.test.js:155]: also a good test!

**Review by nc-nathan-blanch — COMMENTED** [review]
I've left some comments - what if you had an article that exists, but no comments exist? Is this a 200 or a 404?

File: __tests__/app.test.js (hunk 1 above)
Comment (ghiffari90) [reply __tests__/app.test.js:132]: I've changed it in another pull request. Here is the link https://github.com/ghiffari90/my-nc-news/pull/10 :)

**Pull request merged**